pkg/storage/mongo/cataloger.go

```diff
@@ -3,34 +3,39 @@ package mongo
 import (
 	"context"
 
-	"github.com/globalsign/mgo/bson"
 	"github.com/gomods/athens/pkg/errors"
 	"github.com/gomods/athens/pkg/paths"
 	"github.com/gomods/athens/pkg/storage"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo/options"
 )
 
 // Catalog implements the (./pkg/storage).Cataloger interface
 // It returns a list of modules and versions contained in the storage
 func (s *ModuleStore) Catalog(ctx context.Context, token string, pageSize int) ([]paths.AllPathParams, string, error) {
 	const op errors.Op = "mongo.Catalog"
-	q := bson.M{}
+	q := bson.E{}
 	if token != "" {
-		q = bson.M{"_id": bson.M{"$gt": bson.ObjectIdHex(token)}}
+		q = bson.E{Key: "_id", Value: bson.M{"$gt": token}}
 	}
 
-	fields := bson.M{"module": 1, "version": 1}
+	projection := bson.M{"module": 1, "version": 1}
+
+	c := s.client.Database(s.db).Collection(s.coll)
 
-	c := s.s.DB(s.d).C(s.c)
 	modules := make([]storage.Module, 0)
-	err := c.Find(q).
-		Select(fields).
-		Sort("_id").
-		Limit(pageSize).
-		All(&modules)
+	cursor, err := c.Find(context.Background(), q, &options.FindOptions{Projection: projection})
```

Comment:
Can we use `context.WithTimeout` here? 
Like 
```
tctx, cancel := context.WithTimeout(ctx, s.timeout) 
cursor, err := c.Find(tctx, q, &options.FindOptions{Projection: projection})
```
Can you also check other places where context.Background() is used?

Reply:
The only two places where `context.Background()` remains are lines [78](https://github.com/arpitgogia/athens/blob/mongo-go-beta-integration/pkg/storage/mongo/mongo.go#L78) and [57](https://github.com/arpitgogia/athens/blob/mongo-go-beta-integration/pkg/storage/mongo/mongo.go#L57) in mongo.go